### PR TITLE
SapMachine (24): Add dt_filesocket

### PR DESCRIPTION
**Pull request by schmelter-sap**
This is a cherry-pick from sapmachine21 plus adjustments for missing include.,

fixes #1611

### CODE REVIEW

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

File: make/modules/jdk.jdwp.agent/Lib.gmk [hunk 1]
```diff
@@ -47,6 +47,27 @@ TARGETS += $(BUILD_LIBDT_SOCKET)
 ## Build libjdwp
 ################################################################################
 
+# SapMachine 2024-02-16: Add dt_filesocket implementation
+$(eval $(call SetupJdkLibrary, BUILD_LIBDT_FILESOCKET, \
+    NAME := dt_filesocket, \
+    OPTIMIZATION := LOW, \
+    CFLAGS := $(CFLAGS_JDKLIB), \
```
Comment (RealCLanger) [inline make/modules/jdk.jdwp.agent/Lib.gmk]: I think CFLAGS is not needed with latest build system changes

File: make/modules/jdk.jdwp.agent/Lib.gmk [hunk 2]
```diff
@@ -47,6 +47,27 @@ TARGETS += $(BUILD_LIBDT_SOCKET)
...
+    DISABLED_WARNINGS_clang := format-nonliteral, \
```
Comment (RealCLanger) [inline make/modules/jdk.jdwp.agent/Lib.gmk]: Same here, try without DISABLED_WARNINGS_clang

File: make/modules/jdk.jdwp.agent/Lib.gmk [hunk 3]
```diff
@@ -47,6 +47,27 @@ TARGETS += $(BUILD_LIBDT_SOCKET)
...
+    EXTRA_HEADER_DIRS := \
+        include \
+        libjdwp/export, \
+    LDFLAGS := $(LDFLAGS_JDKLIB) \
```
Comment (RealCLanger) [inline make/modules/jdk.jdwp.agent/Lib.gmk]: Try without LDFLAGS

File: make/modules/jdk.jdwp.agent/Lib.gmk [hunk 4]
```diff
@@ -47,6 +47,27 @@ TARGETS += $(BUILD_LIBDT_SOCKET)
...
+        $(call SET_SHARED_LIBRARY_ORIGIN), \
+    LIBS_windows := $(JKDLIB_LIBS) ws2_32.lib Advapi32.lib, \
```
Comment (RealCLanger) [inline make/modules/jdk.jdwp.agent/Lib.gmk]: I'm sure you can remove $(JKDLIB_LIBS). Also not sure about Advapi32.lib ?

File: make/modules/jdk.jdwp.agent/Lib.gmk [hunk 5]
```diff
@@ -47,6 +47,27 @@ TARGETS += $(BUILD_LIBDT_SOCKET)
...
+))
+
+$(BUILD_LIBDT_FILESOCKET): $(call FindLib, java.base, java)
```
Comment (RealCLanger) [inline make/modules/jdk.jdwp.agent/Lib.gmk]: don't think you need this line...

**Review by RealCLanger — CHANGES_REQUESTED** [review]
A few comments regarding the make file.

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

File: make/modules/jdk.jdwp.agent/Lib.gmk (hunk 4 above)
Comment (RealCLanger) [reply make/modules/jdk.jdwp.agent/Lib.gmk]: This still needs to be checked.

File: make/modules/jdk.jdwp.agent/Lib.gmk (hunk 5 above)
Comment (RealCLanger) [reply make/modules/jdk.jdwp.agent/Lib.gmk]: What about that?

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

File: make/modules/jdk.jdwp.agent/Lib.gmk [hunk 6]
```diff
@@ -47,6 +47,22 @@ TARGETS += $(BUILD_LIBDT_SOCKET)
...
 
+# SapMachine 2024-02-16: Add dt_filesocket implementation
+$(eval $(call SetupJdkLibrary, BUILD_LIBDT_FILESOCKET, \
+    NAME := dt_filesocket, \
+    OPTIMIZATION := LOW, \
+    DISABLED_WARNINGS_clang_fileSocketTransport.c := format-nonliteral, \
+    EXTRA_HEADER_DIRS := \
+        include \
+        libjdwp/export, \
+    LIBS_windows := $(JKDLIB_LIBS) ws2_32.lib Advapi32.lib, \
```
Comment (RealCLanger) [inline make/modules/jdk.jdwp.agent/Lib.gmk]: Please remove $(JKDLIB_LIBS) here. Then I think it's good :)

File: make/modules/jdk.jdwp.agent/Lib.gmk (hunk 4 above)
Comment (schmelter-sap) [reply make/modules/jdk.jdwp.agent/Lib.gmk]: Advapi32 ist still needed for permission checking.

File: make/modules/jdk.jdwp.agent/Lib.gmk (hunk 5 above)
Comment (schmelter-sap) [reply make/modules/jdk.jdwp.agent/Lib.gmk]: And it's gone

File: make/modules/jdk.jdwp.agent/Lib.gmk (hunk 3 above)
Comment (schmelter-sap) [reply make/modules/jdk.jdwp.agent/Lib.gmk]: Done

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Review by RealCLanger — APPROVED** [review]
OK, I think that's good then 😄

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Comment by MBaesken** [comment]
Please fix the typo 'doamin socket' .
Otherwise looks okay to me.

**Comment by RealCLanger** [comment]
> Please fix the typo 'doamin socket' . Otherwise looks okay to me.

Done.

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Review by MBaesken — APPROVED** [review]

**Pull request merged**